src/logic/referral/referral.ts

```diff
@@ -167,10 +176,62 @@ export async function createReferralComponent(
       }
 
       const referral = await referralDb.createReferral({ referrer, invitedUser, invitedUserIP })
+
+      const recentInvitations = await referralDb.findReferralProgress({
+        referrer,
+        limit: 2
+      })
+
+      let timeDifference: number | null = null
+      let newestCreatedAt: number | null = null
+      let previousCreatedAt: number | null = null
+
+      if (recentInvitations.length >= 2) {
+        newestCreatedAt = recentInvitations[0].created_at
+        previousCreatedAt = recentInvitations[1].created_at
+        timeDifference = newestCreatedAt - previousCreatedAt
```

Comment:
NIT: we could move the `timeDifference < REFERRAL_FIVE_MINUTES_IN_MS` conditional inside this one

Reply:
I wanted to avoid nested code, but I can do it